tests/common/snappi/read_pcap.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import struct
+import dpkt
+from dpkt.utils import mac_to_str
+
+logger = logging.getLogger(__name__)
+
+PFC_MAC_CONTROL_CODE = 0x8808
+SAMPLE_SIZE = 100000
+UTIL_THRESHOLD = 0.8
+PFC_DEST_MAC = "01:80:c2:00:00:01"
+
+
+def read_pfc_frame(pfc_frame_bytes):
+    """
+    Read PFC frame bytes and return the components of the frame, specifically,
+    the CBFC opcode, the class enable vector, and the class pause times.
+
+    Args:
+        pfc_frame_bytes (bytes): bytes of PFC frame
+
+    Returns:
+        cbfc_opcode (int): CBFC opcode
+        class_enable_vec (int): class enable vector
+        class_pause_times (list of ints): class pause times
+    """
+    cbfc_opcode = struct.unpack(">H", pfc_frame_bytes[0:2])[0]
+    # print("CBFC Opcode:", cbfc_opcode)
+    class_enable_vec = struct.unpack(">H", pfc_frame_bytes[2:4])[0]
+    # print("Class Enable Vector:", class_enable_vec)
+    class_pause_times = []
+    for i in range(0, 16, 2):
+        class_pause_times.append(struct.unpack(">H", pfc_frame_bytes[i + 4:i + 6])[0])
+    # print("Class Pause Times:", [x for x in class_pause_times])
+
+    return cbfc_opcode, class_enable_vec, class_pause_times
+
+
+def validate_pfc_frame(pfc_cap_file):
+    """
+    Validate PFC frame by checking the CBFC opcode, class enable vector and class pause times.
+
+    Args:
+        pfc_cap: PFC frame bytes
+
+    Returns:
+        True if valid PFC frame, False otherwise
+    """
+    f = open(pfc_cap_file, "rb")
+    pcap = dpkt.pcapng.Reader(f)
+
+    curPktCount = 0
+    curPFCPktCount = 0
+    for _, buf in pcap:
+        if curPktCount >= SAMPLE_SIZE:
```

Comment:
it seems that we should pass sample_size as an argument to this function and give it a default value